### DISCUSSION TITLE
Relation between noisepy and noisepy-io?

### DISCUSSION THREAD

**Discussion by jojomale** [mdenolle/NoisePy#322] [Q&A] [answered]
Hi,
Could someone explain to me the structure of the noisepy-package and the relation between the two repositories noisepy and noisepy-io? It seems that the noisepy repo is more active, hosts the documentation and example notebooks. However, the notebooks seem to use the package structure of noisepy-io since they all load from the module noisepy.seis.io, which, to my understanding, is nowhere defined in the noisepy repo. They work when I installed noisepy from PyPI.  When I tried to install from the cloned repository, they did not work.
Thanks for your help.

**Comment by niyiyu** [comment, marked as answer]
Hi @jojomale sorry just found this discussion. 

noisepy-io was previously part of noisepy main package, until we found that several basic IO operations (e.g., reading the stacked correlation, plotting, etc.) don't really need the entire noisepy dependencies. However, in order to use noisepy as a single package, we still need to all dependencies. That's why we decided to make the IO part a separate package: small with very few dependencies. We also expect that people only change the IO part when working on different data, leaving the main package untouched. 

noisepy-io is now a dependent of noisepy, which is why it is automatically installed through PyPI. However, since it shares the same name space with noisepy (noisepy.seis), this makes developing a little bit different.

After cloning the noisepy repo, I usually do `pip install -e .` and manually add the io package into the `src/noisepy/seis/` folder. Unfortunately, doing `pip install .` for two repo, separately, won't work because pip will overwrite the other, makes it no way to merge them into the same folder. 

Best,
Yiyu

**Reply by Kevinmba** [reply]
Hello @niyiyu @
I have done some work with noisepy but i notice the MPI_CLI does not work properly with local ASDF data.
I check other tutorial but none show us how to proceed with ASDF local data.
Please can you check this issue?

**Comment by Kevinmba** [comment]
@niyiyu Hello Please i have run the CLI and jupyter notebook for cross-correlations here is the error.
<img width="1058" height="287" alt="Image" src="https://github.com/user-attachments/assets/be98250c-a97f-4acf-86e5-1929c0671bb9" />
code of config CLI:
acorr_only: false
cc_len: 3600
cc_method: xcorr
channels: [BH?]
client_url_key: IRIS
correction: false
correction_csv: null
down_list: false
end_date: '2007-01-31T00:00:00Z'
freq_norm: rma
freqmax: 8.0
freqmin: 0.05
inc_hours: 24
keep_substack: false
lamax: 6.0
lamin: 2.0
lomax: 11.0
lomin: 8.0
max_over_std: 10
maxlag: 300
ncomp: 3
networks: [XB]
respdir: null
rm_resp: inv
rm_resp_out: VEL
rotation: true
sampling_rate: 20.0
single_freq: true
smooth_N: 10
smoothspect_N: 10
stack_method: all
start_date: '2005-01-01T00:00:00Z'
stations: ['*']
stationxml: false
step: 1800.0
storage_options: {}
substack: true
substack_windows: 24
time_norm: one_bit
xcorr_only: true

error: AttributeError: 'NoneType' object has no attribute 'start_datetime'
